lib/AST/ASTImporter.cpp

```diff
@@ -8013,8 +8013,11 @@ ASTImporter::Import(NestedNameSpecifierLoc FromNNS) {
 
     case NestedNameSpecifier::TypeSpec:
     case NestedNameSpecifier::TypeSpecWithTemplate: {
+      SourceLocation ToTLoc;
+      if (Error Err = importInto(ToTLoc, NNS.getTypeLoc().getBeginLoc()))
```

Comment:
I am not sure if the `getBeginLoc` is the correct location to use here.
Some valid location must be passed to `getTrivialTypeSourceInfo` otherwise the `TSI->getTypeLoc()` returns invalid location.

Reply:
I think any time we have questions like this we should check what the clang parser does.

Reply:
`getBeginLoc` seems good to me.

Reply:
It should work at least in simple case, the source location imported in the new test case looks correct.